src/main/java/org/grouplens/grapht/Instantiators.java

```diff
@@ -39,6 +42,12 @@ private Instantiators() {}
      * @param inst The instance to return (must be non-null).
      * @return An instantiator that returns {@code inst}.
      */
+
+
+    public static final Logger     Logger = LoggerFactory.getLogger(Instantiator.class);
+    private static final LogContext mdcContext = LogContext.create();
```

Comment:
We need to declare a fresh local context every time we use a context; a field (static or otherwise) won't work. This is because creating the context captures the current context value map, and closing/releasing restores it. We can't capture the map until right before the code is running.